x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/ApiKeyService.java

```diff
@@ -1859,12 +1887,25 @@ private void indexInvalidation(
                     apiKeyIdsToInvalidate.add(apiKeyId);
                 }
             }
-            assert false == apiKeyIdsToInvalidate.isEmpty() || false == crossClusterApiKeyIdsToSkip.isEmpty();
+            assert false == apiKeyIdsToInvalidate.isEmpty() || false == crossClusterApiKeyIdsToSkip.isEmpty()
+                : "There are no API keys but that should never happen, original=["
+                    + apiKeys
```

Comment:
I think this could be large (many thousands of API keys) -- I'm not sure we want to log all of that even in the context of tripped assertion. Maybe just log the length of apiKeys? I know that's less useful, just concerned about a humongous entry here.